sources/driverdog/src/tests/copying.conf

```
@@ -0,0 +1,14 @@
+[copying-driver]
+lib-modules-path = "kernel/drivers/extra/copying"
+copy-modules-source = "/usr/share/factory/copying-driver"
+
+[copying-driver.kernel-modules."main.ko"]
+link-objects = []
```

I would probably structure it like this in order to end up with different types:
```suggestion
[copying-driver]
lib-modules-path = "kernel/drivers/extra/copying"

[copying-driver.kernel-modules."main.ko"]
copy-source = "/usr/share/factory/copying-driver"
```

Corresponding Rust code might look like:
```
enum DriverType {
    Linking(LinkingDriver),
    Copying(CopyingDriver),
}

struct LinkingDriver {
    name: String,
    link_objects: Vec<String>,
}

struct CopyingDriver {
    name: String,
    copy_source: PathBuf
}
```

That way, you can match on the different types and code just the behavior you need for that type, and you can avoid the need for an empty link-objects list.

---

I'll give this a shot. I think I'll need to update a bit on where `objects-source` is located to make this really consistent but should make the types more helpful for tracking what work is expected to be done.

---

I've moved to an Enum to provide this type of switching.